Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/ContactDiarySettings.kt

```diff
@@ -1,27 +1,20 @@
 package de.rki.coronawarnapp.contactdiary.ui
 
-import android.content.Context
-import de.rki.coronawarnapp.util.di.AppContext
-import de.rki.coronawarnapp.util.preferences.clearAndNotify
-import de.rki.coronawarnapp.util.preferences.createFlowPreference
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class ContactDiarySettings @Inject constructor(
-    @AppContext val context: Context
-) {
+class ContactDiarySettings @Inject constructor(val preferences: ContactDiaryPreferences) {
 
-    private val prefs by lazy {
-        context.getSharedPreferences("contact_diary_localdata", Context.MODE_PRIVATE)
-    }
-
-    val isOnboarded = prefs.createFlowPreference(
-        key = "contact_diary_onboarded",
-        defaultValue = false
-    )
+    var onboardingStatus: OnboardingStatus
+        get() {
+            val order = preferences.onboardingStatusOrder.value
+            return OnboardingStatus.values().find { it.order == order } ?: OnboardingStatus.NOT_ONBOARDED
+        }
+        set(value) = preferences.onboardingStatusOrder.update { value.order }
 
-    fun clear() {
-        prefs.clearAndNotify()
+    enum class OnboardingStatus(val order: Int) {
+        NOT_ONBOARDED(-1),
+        RISK_STATUS_1_12(0)
```

Comment:
If I change this to `RISK_STATUS_1_12(**0**)` it works nicely. I suppose retrieving the data from the preferences returns 0 as the default value at some point.

Reply:
@ralfgehrer thanks!

Reply:
fixed it